source/vstd/std_specs/vecdeque.rs

```diff
@@ -0,0 +1,374 @@
+/// This code adds specifications for the standard-library type
+/// `std::collections::VecDeque`.
+use super::super::prelude::*;
+
+use core::alloc::Allocator;
+use core::clone::Clone;
+use core::option::Option;
+use core::option::Option::None;
+use std::collections::vec_deque::Iter;
+use std::collections::VecDeque;
+use std::ops::Index;
```

Comment:
I believe this set of changes would allow it to work on both std and no-std `alloc` contexts (thereby allowing the `#[cfg(all(feature = "alloc", feature = "std"))]` to be de-restricted to just `#[cfg(feature = "alloc")]`)
```suggestion
use alloc::collections::vec_deque::Iter;
use alloc::collections::vec_deque::VecDeque;
use core::ops::Index;
```

Reply:
Thanks! I'll try that. Presumably this approach could also be applied to `HashMap` and `HashSet`?

Reply:
Oh, it looks like `HashMap` and `HashSet` aren't in `alloc`.

Reply:
Yeah, them not being in `alloc` and being restricted to `std` is a long-standing open issue: https://github.com/rust-lang/rust/issues/27242